### DISCUSSION TITLE
[domurl] Incorrect type for Url.paths

### DISCUSSION THREAD

**Discussion by besfahbod** [DefinitelyTyped/DefinitelyTyped#55760] [Issues with a @types package]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a61d1d2ebb71895d357fe19543fb0ee9dfe613b1/types/domurl/index.d.ts#L25

This test code passes at runtime:
```
const url = new Url("/my/site/doc/path?foo=bar#baz");
expect(String(url)).toEqual("/my/site/doc/path?foo=bar#baz");
expect(url.path).toEqual("/my/site/doc/path");
expect(url.paths()).toEqual(["my", "site", "doc", "path"]);
expect(url.paths()[2]).toEqual("doc");
expect(url.paths()[5]).toEqual(undefined);
```

However, causes these type-check errors:
```
[...] - error TS2493: Tuple type '[string]' of length '1' has no element at index '2'.

[...]     expect(url.paths()[2]).toEqual("doc");
                          ~

[...] - error TS2493: Tuple type '[string]' of length '1' has no element at index '5'.

[...]     expect(url.paths()[5]).toEqual(undefined);
                          ~
```

I believe the output type of the `paths()` method needs to be updated to "array of strings", instead of a single-element tuple.

**Comment by besfahbod** [comment]
We probably should add unit test for `url.paths` to https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/domurl/domurl-tests.ts

[a comment by an automated account is not shown]

**Comment by besfahbod** [comment]
@orta, could you please clarify what was the reason to close https://github.com/DefinitelyTyped/DefinitelyTyped/issues/55642 and make this Discussion item?

I'm pretty sure of the incorrect typing provided, and have unit test verifying the expected behavior. This sounds like a clear bug report (issue) to me.

Am I missing something here?

**Reply by orta** [reply]
Yep, all DT types discussion happens in discussions: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377 so I moved this over

**Reply by besfahbod** [reply]
Thanks, orta! That explains it well.
